v2/tools/generator/internal/jsonast/swagger_type_extractor.go

```diff
@@ -552,9 +578,43 @@ func (extractor *SwaggerTypeExtractor) expandEnumsInPath(
 		operationPath,
 	}
 
+	_, subpath, err := extractor.extractResourceSubpath(operationPath)
+	if err != nil {
+		// This error is safe to ignore in this case as it means that we can't parse the path as a resource.
+		// Just return the raw path so we do further processing and log a message
+		klog.Errorf("Error expanding enums in path (%s): %s", extractor.swaggerPath, err.Error())
+		return results
+	}
+
+	paramMap := make(map[string]spec.Parameter)
 	for _, param := range parameters {
 		_, resolved := extractor.fullyResolveParameter(param)
 
+		paramMap[resolved.Name] = resolved
+	}
+
+	split := strings.Split(subpath, "/")
+	// Loop over split to ensure we're processing the url paths left to right, which is important
+	// for counting which parameters correspond to a "name" and which correspond to a "type"
+	for i, urlPart := range split {
+		if len(urlPart) == 0 {
+			// skip empty parts
+			continue
+		}
+
+		if urlPart[0] != '{' {
+			// Not a parameter, so by definition doesn't need to be expanded
+			continue
+		}
+
+		paramName := strings.Trim(urlPart, "{}")
+		param, ok := paramMap[paramName]
```

Comment:
Does Swagger require things to match by case? Or should this be a case-insensitive lookup?